File: languages/go/entity/identityV2/integer_attribute_descriptor.go
package identityV2

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

type IntegerAttributeDescriptor struct {
	AttributeDescriptor
}

func NewIntegerAttributeDescriptor(name string, id string, description string, required bool) IntegerAttributeDescriptor {
	a := AttributeDescriptor{
		DisplayName: name,
		Id:          id,
		Description: description,
		Required:    required,
	}
	return IntegerAttributeDescriptor{
		AttributeDescriptor: a,
	}
}

func NewNumberAttributeDescriptorFromProto(s *proto.IntegerAttributeDefinitionV2) IntegerAttributeDescriptor {
	if s == nil {
		return IntegerAttributeDescriptor{}
	}
	a := AttributeDescriptor{
		DisplayName: s.GetId(),
		Id:          s.Id,
		Description: s.Description,
		Required:    s.Required,
	}
	return IntegerAttributeDescriptor{
		AttributeDescriptor: a,
	}
}

func (s IntegerAttributeDescriptor) ToProto() *proto.IntegerAttributeDefinitionV2 {
	return &proto.IntegerAttributeDefinitionV2{
		DisplayName: s.DisplayName,
		Id:          s.Id,
		Description: s.Description,
		Required:    s.Required,
	}
}

File: languages/java/src/main/java/com/bloock/sdk/entity/identity/IntegerAttributeDescriptor.java
package com.bloock.sdk.entity.identity;

import com.bloock.sdk.bridge.proto.BloockIdentityEntities;

/**
 * Represents a descriptor for an attribute with an integer value.
 */
public class IntegerAttributeDescriptor extends AttributeDescriptor {
  /**
   * Constructs an IntegerAttributeDescriptor object with the specified
   * parameters.
   * 
   * @param displayName
   * @param technicalName
   * @param description
   * @param required
   */
  public IntegerAttributeDescriptor(
      String displayName, String technicalName, String description, Boolean required) {
    super(displayName, technicalName, description, required);
  }

  public static IntegerAttributeDescriptor fromProto(
      BloockIdentityEntities.IntegerAttributeDefinition res) {
    return new IntegerAttributeDescriptor(
        res.getDisplayName(), res.getId(), res.getDescription(), res.getRequired());
  }

  public BloockIdentityEntities.IntegerAttributeDefinition toProto() {
    return BloockIdentityEntities.IntegerAttributeDefinition.newBuilder()
        .setDisplayName(this.displayName)
        .setId(this.technicalName)
        .setDescription(this.description)
        .setRequired(this.required)
        .build();
  }
}
